### ISSUE TITLE
Gentoo support

### ISSUE THREAD

**Issue by chicken2006** [fcorbelli/zpaqfranz#149]
Will this project be supported on Gentoo?

**Comment by fcorbelli** [comment]
I will try just today 😄

**Comment by fcorbelli** [comment]
Creating a Gentoo virtual machine requires an incredibly large amount of time, much more than I expected. Definitely much more.

**Comment by chicken2006** [comment]
Yeah, setting up Gentoo requires you a lot of time. If you are impatient, you can use binary version instead of compiling.

**Comment by chicken2006** [comment]
Do you have time for setting up the VM?

**Comment by fcorbelli** [comment]
I am trying

**Comment by fcorbelli** [comment]
![nobig](https://github.com/user-attachments/assets/e1f5743c-a841-4ed2-967f-477e919155b8)

Done
No particular problem founded compiling with g++

I'm studying how to submit packages to Gentoo's official repository. 

It took me three years to get one accepted for Debian 😄

**Comment by fcorbelli** [comment]
E-mail sended to Gentoo's zpaq mantainer.
We'll see

**Comment by chicken2006** [comment]
Has there been any progress?

**Comment by fcorbelli** [comment]
Nope, currently no answer
You can try yourself asking this guy

https://packages.gentoo.org/maintainer/mgorny@gentoo.org

**Comment by TimovVeen** [comment]
For the time being I use this simple _zpaqfranz-60.10.ebuild_ in a local repository. I will put it here in case it is useful for someone else:
```bash
# Copyright 1999-2025 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=8

DESCRIPTION="Journaling incremental deduplicating archiving compressor"
HOMEPAGE="https://github.com/fcorbelli/zpaqfranz"
SRC_URI="https://github.com/fcorbelli/zpaqfranz/archive/refs/tags/${PV}.tar.gz"

LICENSE="Unlicense"
SLOT="0"
KEYWORDS="~amd64 ~arm64 ~x86"

src_install() {
	dobin "./zpaqfranz"
}
```

**Comment by chicken2006** [comment]
I use this zpanqfranz-9999.ebuild:
`# Copyright 1999-2024 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=8

inherit flag-o-matic pax-utils toolchain-funcs

DESCRIPTION="Deduplicating archiver with encryption and paranoid-level tests"
HOMEPAGE="https://github.com/fcorbelli/zpaqfranz"

LICENSE="MIT"
SLOT="0"
IUSE="debug nightly"

if [[ ${PV} == *9999* ]]; then
	EGIT_REPO_URI="https://github.com/fcorbelli/zpaqfranz.git"
	SRC_URI="http://www.francocorbelli.it/zpaqfranz.cpp"
	inherit git-r3
else
	SRC_URI="https://github.com/fcorbelli/zpaqfranz/archive/refs/tags/${PV}.tar.gz
	nightly? ( http://www.francocorbelli.it/zpaqfranz.cpp )
	"
	KEYWORDS="amd64 ~arm64 x86"
fi

# perl for pod2man
BDEPEND="dev-lang/perl"

src_prepare() {
	eapply_user
	cp "${DISTDIR}"/zpaqfranz.cpp ./
}

src_compile() {
	use debug || append-cppflags -DNDEBUG
	emake CXX="$(tc-getCXX)" CXXFLAGS="${CXXFLAGS}"
}

src_test() {
	pax-mark m zpaqfranz
	default
}

src_install() {
	emake install BINDIR="${ED}"/usr/bin
	pax-mark m "${ED}"/usr/bin/zpaqfranz
	doman man/zpaqfranz.1
	dodoc man/zpaqfranz.pod
	einstalldocs
}`

**Comment by fcorbelli** [comment]
This is not good http://www.francocorbelli.it/zpaqfranz.cpp

Because this is a temporary pre-prelease that I use for developing
Sometime does not compile at all

**Comment by chicken2006** [comment]
I know that, so I use the "nightly" USE flag to compile this unstable cpp file only when it's enabled. By the way, the zpaqfranz-9999.ebuild is already unstable ebuild.
